File: vol1/ch8/LeetCode414.py
from sortedcontainers import SortedSet
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        tset=SortedSet()						#默认递增排列
        for x in nums:
            tset.add(x)
        if len(tset)<3:
            return tset[-1]
        else:
            return tset[-3]

File: vol1/ch8/LeetCode414.cpp
struct Cmp {
    bool operator()(const int&a,const int&b) const {
        return a>b;         //指定按元素值递减排列
    }
};
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        set<int,Cmp> tset;
        for(int x:nums)
            tset.insert(x);
        auto it=tset.begin();
        if(tset.size()>=3) {
            it++; it++;
        }
        return *it;
    }
};

